master/internal/task/allocation.go

```diff
@@ -165,6 +167,12 @@ func (a *Allocation) Receive(ctx *actor.Context) error {
 		if err := a.ResourcesAllocated(ctx, msg); err != nil {
 			a.Error(ctx, err)
 		}
+
+		for cID := range a.reservations {
+			prom.AssociateAllocationTask(a.req.AllocationID.String(), a.req.TaskID.String())
+			prom.AddAllocationContainer(a.reservations[cID].Summary())
```

Comment:
so re comment above, this would become `AddAllocationReservation` and know how to handle the stuff happening in agents.go too.